# Практическое задание к 1 семинару (Расчет вероятности случайных событий)
## Задание 1
Из колоды в 52 карты извлекаются случайным образом 4 карты. 

### a) Найти вероятность того, что все карты – крести.

$P(A) = \frac{ 13 }{ 52 }*\frac{ 12 }{ 51 }*\frac{ 11 }{ 50 }*\frac{ 10 }{ 49 } = 0.002641$

**Ответ: $P(A) = 0.0026410564225690276$ или $0.26$%**

### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

n = 52 - карт в колоде

t = 4  - колличество тузов

a = 4 - количество извлекаемых карт

1) находим по формуле сочетания количество благоприятных комбинайци. $$C_n^{k} = \frac{n!}{k!(n-k)!}$$ 

берём подмножество тузов $$C_4^{1} = \frac{4!}{1!(4-1)!} = 4$$ 

берём подмножество без тузов $$C_{48}^{3} = \frac{48!}{3!(48-3)!} = 17296$$

находим количество благоприятных исходов $$C_A = C_4^{1} *  C_{48}^{3} = 4 * 17296 = 69184$$ 



In [1]:
from math import factorial as f

In [2]:

cards = 52  # количество карт
ace = 4   # количество тузов
need_ace = 1   # необходимо тузов
others_cards = 3   # необходимо других карт
a= 4    # количество извлекаемых карт

# берём подмножество тузов
tuz = f(ace)/(f(need_ace)*f(ace-need_ace))
# берём подмножество без тузов
notuz = f(cards-ace)/(f(others_cards)*f(cards-ace-others_cards))
#   перемножаем результатыf
res = tuz*notuz # количество благоприятных исходов
print("tuz = ", tuz, "\nnotuz = ", notuz, "\nres = ", res)


tuz =  4.0 
notuz =  17296.0 
res =  69184.0


2) считаем общее число исходов

In [3]:
allPos = f(cards)/(f(a)*f(cards-a))
print("allPos = ",allPos)

allPos =  270725.0


In [4]:
p = res/allPos  # ситаем вероятность
p

0.25555083571890297

$$P_{(A)} = \frac{69184}{270725} = 0.25555083571890297$$ 
**Ответ: $P(A) = 0.25555083571890297$ или $25.5$%**

## Задание 2
На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. 

Код содержит три цифры, которые нужно нажать одновременно.

Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

посчитаем количество сочетаний :
$$C_{n}^{k} = \frac{n!}{к!(n-k)!}$$ 

In [5]:
button = 10  # количество кнопок
num = 3   # количество кнопок в комбинации

# находим количество возможных вариантов
var = f(button)/(f(num)*f(button-num))
print("var = ", var)


var =  120.0


$$C_{10}^{3} = \frac{10!}{3!(10-3)!} = 120$$ 
**Ответ: $P(A) = \frac{ 1 }{ 120 }$**

## Задача 3
В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали.

Какова вероятность того, что все извлеченные детали окрашены?

$P(A) = \frac{ 9 }{ 15 }*\frac{ 8 }{ 14 }*\frac{ 7 }{ 13 } = \frac{ 12 }{ 65 } = 0,1846$

### Задача 4
В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

**Ответ: $P(A) = \frac{ 2 }{ 100 }*\frac{ 1 }{ 99} = \frac{ 1 }{ 4950 }$**